### DISCUSSION TITLE
How to support hacktribe.

### DISCUSSION THREAD

**Discussion by bangcorrupt** [bangcorrupt/hacktribe#63] [Announcements]
Hacktribe is free (as in beer) and always will be.  By testing Hacktribe you help find bugs, improving the user experience for everyone.  

If you want to support this project financially, you are most welcome to [become a sponsor](https://github.com/sponsors/bangcorrupt).  If you would like to make a crypto donation, post a comment with your preferred currency and I'll set up an address.  In any month where you sponsor $10 or more you will have access to a private repository, where you can vote for the next feature update and receive priority support.  If you don't receive an invite automatically, leave a message in this thread and I'll fix it.

I have food and shelter covered for now, so any funds received will go directly to improving Hacktribe.   I'm slowly saving for a KingKorg, as it has the same sound engine with more of the features included.  Dissecting the sound engine could provide more oscillator types, filter models and fx devices.  

Also, search for anything labeled `good first issue` in [discussions](https://github.com/bangcorrupt/hacktribe/discussions?discussions_q=label%3A%22good+first+issue%22) and [issues](https://github.com/bangcorrupt/hacktribe/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).  These should provide opportunities to contribute to Hacktribe without hacking the firmware.

Thank you for supporting Hacktribe!

**Comment by BKLronin** [comment]
First! :)

**Reply by bangcorrupt** [reply]
Thank you so much! :1st_place_medal:

**Comment by maks** [comment]
You've done a fantastic job already on this project! very happy to chip in a little bit to help you keeping going.

**Reply by bangcorrupt** [reply]
Incredibly generous of you, thank you very much!

**Reply by bangcorrupt** [reply]
Thank you so much for your continued support!  Next time I see a Mini Kaoss Pad 2s going cheap I'll grab it and dissect the FX engine.

**Comment by bangcorrupt** [comment]
Thank you again to everyone who has sponsored hacktribe! :smile: 

Next month I will buy  [ADZS-ICE-1000](https://www.digikey.co.uk/en/products/detail/ADZS-ICE-1000/ADZS-ICE-1000-ND/5175780) debug adapter for the Blackfin DSP.  This will hopefully provide some insight into the workings of the sound engine.  Reverse engineering the communication protocol between the CPU and DSP is a major step on the way to an open-source firmware.

**Reply by hobbowizard** [reply]
wow so what will this allow you to do? what are the goals for hacktribes future?

**Reply by bangcorrupt** [reply]
#157

**Comment by hobbowizard** [comment]
how much would it cost me to get someone to update the patching instructions so it actually makes sense for windows users? I will get you a beer bro!

**Reply by bangcorrupt** [reply]
You can't buy a specific feature, this is something I do for fun in my free time.  In general, I will spend more time trying to help people who sponsor the project, but there's never a guaranteed result.

**Reply by hobbowizard** [reply]
@bangcorrupt hey dude sorry about that thanks for deleting those comments

please may I ask you, is there anyway or will you be adding any way to cut up a sample using time  slice and put across the keys of a single part?

**Reply by bangcorrupt** [reply]
If I can get #27 working at some point then it shouldn't be too difficult to key map sample slices.  Key tracking as a modulation source is also planned; this could modulate sample start time.

By the way, I deleted the instructions you posted by accident, you're welcome to post these again in #41 if you want to.  Many of your other comments were unhelpful and some were abusive.

**Comment by djokenzi** [comment]
I made a little donation - it's not much, but I hope it brings you a little bit closer to your goal. Keep up the good work. I really hope we'll see all these great new features on E2 soon!

**Reply by bangcorrupt** [reply]
Thank you so much! :smile:

The next update will be FX editing.  It's already partly possible using NRPN, but it's mostly undocumented, not user friendly and we can't save anything yet.  Displaying things on the screen is proving difficult, so I may make an external editor for setting up FX presets.

**Comment by f-galland** [comment]
Do you receive crypto donations? My country has restrictions on international payments. Thanks!

**Reply by bangcorrupt** [reply]
Hi, I'm happy to accept donations in any currency I can.  I have a coinbase account if that works for you.  Which cryptocurrency would you like to use?

**Reply by f-galland** [reply]
I think coinbase will work. Maybe doing it via USDT if possible.
I'd need the wallet ID and the network. Thanks!

**Reply by bangcorrupt** [reply]
USDT address:
0xA9ABbA64e51907D16D04836E13366Fbaec66A60b

It says "Tether (ERC-20) This is an ERC-20 (Ethereum) address."  Is that the network?

**Reply by f-galland** [reply]
That seems valid. It should have gone through. Been meaning to do this for a long time!

Regards!

**Reply by bangcorrupt** [reply]
That's gone through, thank you very much :smiley:  !

**Reply by BarBryzze** [reply]
Is that address still valid? I'm living on a budget but I have a tiny amount of bitcoin I don't use and has been sitting in my coinbase acount since I bought it. I'd like to send it to you, it'll buy you some beers at least.

Awesome work. Thanks a lot for the Windows GUI, I was afraid of doing something wrong so besides checking this page when I have time I hadn't tried to install Hacktribe yet. It worked like a charm on my e2s. Now in doubt if I should install it on my e2 as well. Anyway, thank you a lot for doing this.

**Reply by bangcorrupt** [reply]
Hi, you're very welcome.  If your crypto is in BTC it is probably better to use a Bitcoin address.

3CXsiwCHs4enhxkKJESL3twaWLjNcYYoiL

Coinbase says: _"This is a bitcoin network address. Do not send BTC over any other network to this address or your funds may be lost. Do not send Bitcoin Cash (BCH) to this address."_

> Now in doubt if I should install it on my e2 as well.

The synth firmware has some features that aren't yet available in Hacktribe, like multi-sampled instruments and a drive stage for the PCM samples.

> I'm living on a budget

I think Coinbase makes it quite easy to sell once you have verified your ID.

**Reply by bangcorrupt** [reply]
That's gone through, thank you.  I sent you an invite to the sponsors discussions.

**Reply by BarBryzze** [reply]
I think that worked. I'm surprised I was able to log in to that account at all, hope you received it. 
Yeah I think I'll leave the synth as is for now, actually just got these units only half a year ago so I'm still figuring stuff out, they're also my first pieces of gear so I'm starting from zero, and now it got even more interesting. 

Only one question comes to mind right now, would it be possible to have feedback on which notes are being played when using the keyboard? I'm usually using Reaper or Midiox to connect to a usb/midi keyboard but it would be nice to have that information on the display of the e2s. 
I have no idea how much work you're putting into this but it defenitely made my day(s).

**Reply by BarBryzze** [reply]
Thx!

**Reply by bangcorrupt** [reply]
> feedback on which notes are being played when using the keyboard

I haven't made much progress printing to the display within Hacktribe yet, but I got an LCD driver working in a fully custom firmware, so this should be possible some day.

**Reply by f-galland** [reply]
Hi!

Is this still a valid address?

> USDT address:
> 0xA9ABbA64e51907D16D04836E13366Fbaec66A60b

Let me know!
Regards

**Reply by bangcorrupt** [reply]
I'm not sure.  I created a new one just in case.

USDT:
0xAB2d34555f453A4691398c14CA0808e99F62b5fE

**Comment by bukru** [comment]
Are you alone to make all the work ?
What is your job ? Dev ?
I would like to participate to dev some feature but i'm not sure it is possible.
In your opinion what are the minimum requirements to try somethint ?

I'll made little financial help too.

Have a good day

**Reply by bangcorrupt** [reply]
Just me hacking the firmware, a few people have contributed scripts and content.  

I'm a student, I study music and computing.  

All you really need is stubborn persistence and the willingness to fail. I knew very little about reverse engineering when I started Hacktribe, my coding has improved a lot by working on it.

Thank you.

**Reply by bukru** [reply]
Did you start by just hexdump the firmware or do you use a tools to decompile it or something like that ?

**Reply by bangcorrupt** [reply]
I started by hex editing the binary using python, then used Ghidra to decompile and rizin to debug.  There are some notes in #89.  The [debrick guide](https://github.com/bangcorrupt/hacktribe/wiki/Debrick) has some notes on attaching a debugger.  [This script](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/e2-backup-bootloader.py) can dump the bootloader from RAM of running Hacktribe.

**Comment by MoroAlmostDone** [comment]
I installed the firmware this evening, definitely going to donate, wait till i get some money, don't hack my computer xD Great Job!!!!!

**Reply by bangcorrupt** [reply]
> definitely going to donate

Thank you.

> don't hack my computer xD

Not that sort of hacker.

**Reply by MoroAlmostDone** [reply]
Hi, just made my donation, anyway what would be the best option: turning the red one into hacktribe and leaving the blue one normal or viceversa? In your opinion

**Reply by bangcorrupt** [reply]
> best option: turning the red one into hacktribe and leaving the blue one normal

Yes, I would do it like this.  Hacktribe is missing the drive effect the synth firmware has on the Osc Edit knob for PCM samples.  It also doesn't support the multi-sampled pianos in the synth firmware.  Hopefully I can add both of these some day.

Thank you for your support! :smiley:

**Reply by MoroAlmostDone** [reply]
Anyway, i'm not sure if i can a make a better job than you did, but i'm a bit of a pixel artist and if you want i can try to make the "Hacktribe" loading logo, in the electribe font or something different if thats an option

**Reply by bangcorrupt** [reply]
Some new splash screens would be cool.  I'm not great with graphics, it took me ages to bodge up the one we have.  

The image is 128x64, 1 bit per pixel inverted monochrome.  There is a [script for replacing the splash screen](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/ht_splash_screen.py) in a patched binary.  

You need to patch first, then run the script to change the bitmap.  If you want me to test some without risking bricking your electribe, post them in a separate discussion and I'll try them out.
